### PR TITLE
Remove copying env var to .bsp/mill.json

### PR DESCRIPTION
**Pull request by joan38**

### CODE REVIEW

**Review by lefou — COMMENTED** [review]
Thanks for the PR. Sorry for the bikeshedding below, but I looked way too long at this snippet trying to understand why there is a `split.last`.

File: bsp/src/mill/bsp/BSP.scala [hunk 1]
```diff
@@ -64,11 +64,7 @@ object BSP extends ExternalModule {
     write(
       BspConfigJson(
         "mill-bsp",
-        Seq(
-          "sh",
-          "-c",
-          s"env ${sys.env.map { case (k, v) => s""""$k=$v"""" }.toSeq.mkString(" ")} $millPath -i ${BSP.getClass.getCanonicalName.split("\\$").last}/start"
-        ),
+        Seq(s"$millPath -i ${BSP.getClass.getCanonicalName.split("\\$").last}/start"),
```
Comment (lefou) [inline bsp/src/mill/bsp/BSP.scala:67]:
```suggestion
        Seq(s"$millPath -i ${BSP.getClass.getCanonicalName.split("\\$").head}/start"),
```

Although, this is effectively an array of length 1 and both will give the same result, semantically, we want the first part of the split, not the last.

Comment (joan38) [reply bsp/src/mill/bsp/BSP.scala:67]: Good point

**Comment by lefou** [comment]
A run on my own Travis-CI account (https://travis-ci.org/github/lefou/mill/builds/740495671) gave the following error: 

```
[info] Compiling 9 Scala sources to /home/travis/build/lefou/mill/out/bsp/compile/dest/classes ...
[error] /home/travis/build/lefou/mill/bsp/src/mill/bsp/BSP.scala:67:73: value first is not a member of Array[String]
[error] Error occurred in an application involving default arguments.
[error]         Seq(s"$millPath -i ${BSP.getClass.getCanonicalName.split("\\$").first}/start"),
[error]                                                                         ^
```

Method `.first` isn't valid, but `.head` (my initial suggestion) is.

**Comment by joan38** [comment]
Lol, I don't know why I wrote first.
Corrected

**Comment by lefou** [comment]
Thanks!

**Pull request merged**